agent/handlers/types/v2/response.go

```diff
@@ -105,6 +105,24 @@ func NewTaskResponse(taskARN string,
 	return resp, nil
 }
 
+// NewContainerResponse creates a new container response based on container id
+func NewContainerResponse(containerID string,
+	state dockerstate.TaskEngineState) (*ContainerResponse, error) {
+	dockerContainer, ok := state.ContainerByID(containerID)
+	if !ok {
+		return nil, errors.Errorf(
+			"v2 container response: unable to find container '%s'", containerID)
+	}
+	task, ok := state.TaskByID(containerID)
+	if !ok {
```

Comment:
Wouldn't the IP check in `metadataV2Handler` method suffice to check if the container belongs to a task?

Reply:
in theory, yeah. but, we have a boolean that's being returned here it's always good to use that to future proof this.